catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -813,6 +843,179 @@ public void unknownOperation() throws CswException {
+            throws CswException {
+        if (request == null) {
+            throw new CswException("GetRecordsSubscription request is null");
+        } else {
+            LOGGER.debug("{} attempting to subscribe.", request.getRequest());
+        }
+        if (StringUtils.isEmpty(request.getVersion())) {
+            request.setVersion(CswConstants.VERSION_2_0_2);
+        } else {
+            validateVersion(request.getVersion());
+        }
+
+        LOGGER.trace("Exiting getRecordsSubscription");
+
+        return createOrUpdateSubscription(request.get202RecordsType(), null);
+    }
+
+    @Override
+    @POST
+    @Path("/subscription")
+    @Consumes({MediaType.TEXT_XML, MediaType.APPLICATION_XML})
+    @Produces({MediaType.WILDCARD})
+    public Response createRecordsSubscription(GetRecordsType request) throws CswException {
+        return createOrUpdateSubscription(request, null);
+    }
+
+    private Response createOrUpdateSubscription(GetRecordsType request, String requestId)
+            throws CswException {
+
+        if (request == null) {
+            throw new CswException("Request is null");
+        } else {
+            LOGGER.debug("{} attempting to subscribe.", request.getService());
+        }
+
+        validateOutputFormat(request.getOutputFormat());
+
+        validateOutputSchema(request.getOutputSchema());
+
+        if (request.getAbstractQuery() != null) {
+            if (!request.getAbstractQuery()
+                    .getValue()
+                    .getClass()
+                    .equals(QueryType.class)) {
+                throw new CswException("Unknown QueryType: " + request.getAbstractQuery()
+                        .getValue()
+                        .getClass());
+            }
+
+            QueryType query = (QueryType) request.getAbstractQuery()
+                    .getValue();
+
+            validateTypes(query.getTypeNames(), CswConstants.VERSION_2_0_2);
+
+            validateElementNames(query);
+
+            if (query.getConstraint() != null &&
+                    query.getConstraint()
+                            .isSetFilter() && query.getConstraint()
+                    .isSetCqlText()) {
+                throw new CswException("A Csw Query can only have a Filter or CQL constraint");
+            }
+        }
+
+        QueryRequest query = getQuery(request);
+        URI deliveryMethodUrl;
+        if (request.getResponseHandler() != null && !request.getResponseHandler()
+                .isEmpty()) {
+
+            try {
+                deliveryMethodUrl = new URI(request.getResponseHandler()
+                        .get(0));
+            } catch (URISyntaxException e) {
+                throw new CswException("Invalid ResponseHandler URL", e);
+            }
+        } else {
+            throw new CswException("Subscriptions require a ResponseHandler URL to be specified");
+        }
+
+        SendEvent sendEvent = new SendEvent(mimeTypeTransformerManager,
```

Comment:
Could we create the event and subscription somewhere else, e.g., in a factory class or inside the `SubscriptionManager` instead of here?